gprofiler/main.py

```diff
@@ -1072,8 +1095,11 @@ def main() -> None:
             spark_sampler = None
 
         if hasattr(args, "func"):
-            assert args.subcommand == "upload-file"
-            args.func(args, profiler_api_client)
+            if args.subcommand == "extract-resources":
```

Comment:
`extract-resources`, unlike `upload-file` is a subcommand which requires much less preparation/verification of preconditions. By putting its handling this late in the code, you have to fix various sites like
```
    if not args.upload_results and not args.output_dir and not args.extract_resources:
        parser.error("Must pass at least one output method (--upload-results / --output-dir)")
```
and there are also some unfixed sites - for example, `verify_preconditions` still runs and unnecessarily checks for root, etc.

Please move `extract-resources` handling to be earlier, right after `parse_args()`. Also don't forget to move this check:
```
    if args.extract_resources and args.resources_dest == "":
        parser.error("Must provide --resources-dest when extract-resources")
```